src/components/ConfDeletePopup.js

```diff
@@ -0,0 +1,43 @@
+import Popup from "./Popup";
+
+class ConfirmDeletePopup extends Popup {
+  cardId = null;
+  constructor({ popupSelector, handleFormSubmit }) {
+    super(popupSelector);
+    this._popupForm = this.popupElement.querySelector(".popup__form");
+    this.handleFormSubmit = handleFormSubmit;
+    this._setEventListeners();
+  }
+
+  // Event Listeners setter and submit logics
+  _setEventListeners() {
+    this._popupForm.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      if (
+        this.handleFormSubmit &&
+        typeof this.handleFormSubmit === "function"
+      ) {
+        // здесь должна подтягиваться информация о карточке, которая будет удалена
+        this.handleFormSubmit(this.cardId, this.cardElement); // сюда надо передавать информацию о карточке
+        this.setCardId(null);
```

Comment:
`        this.setCardId(null);
`

You need to **clear** the data only when you **open** it or better in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.

Reply:
Done.